File: src/test/java/io/github/subiyacryolite/jds/Serialization.java
package io.github.subiyacryolite.jds;

import io.github.subiyacryolite.jds.common.BaseTestConfig;
import javafx.beans.property.SimpleBlobProperty;
import org.junit.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * Created by indana on 5/10/2017.
 */
public class Serialization extends BaseTestConfig {

    @Test
    public void testSerialization() {
        List<JdsEntity> entities = new ArrayList<>();
        entities.add(getSimpleAddressBook());
        for (JdsEntity jdsEntity : entities) {
            String canonicalName = jdsEntity.getClass().getCanonicalName();
            serialize(jdsEntity, canonicalName);
            deserialize(canonicalName, jdsEntity.getClass());
        }
    }

    @Test
    public void testBlobSerialization() {
        SimpleBlobProperty in = new SimpleBlobProperty(new byte[]{1, 0, 1, 1, 1, 1, 0});
        String canonicalName = in.getClass().getCanonicalName();
        serialize(in, canonicalName);
        SimpleBlobProperty out = deserialize(canonicalName, in.getClass());
        System.out.printf("pre %s\n",Arrays.toString(in.get()));
        System.out.printf("post %s\n",Arrays.toString(out.get()));
    }
}

File: src/test/java/tests/Serialization.kt
package tests

import common.BaseTestConfig
import io.github.subiyacryolite.jds.beans.property.SimpleBlobProperty
import org.junit.jupiter.api.Test

import java.util.Arrays

/**
 * Created by indana on 5/10/2017.
 */
class Serialization : BaseTestConfig("Serialization") {

    @Test
    fun testSerialization() {
        initialiseSqLiteBackend()
        for (jdsEntity in jdsDb.classes.values) {
            val canonicalName = jdsEntity.canonicalName
            serialize(jdsEntity, canonicalName)
            deserialize(canonicalName, jdsEntity)
        }
    }

    @Test
    fun testBlobSerialization() {
        val simpleBlobProperty = SimpleBlobProperty(byteArrayOf(0xC9.toByte(), 0xCB.toByte(), 0xBB.toByte(), 0xCC.toByte(), 0xCE.toByte(), 0xB9.toByte(), 0xC8.toByte(), 0xCA.toByte(), 0xBC.toByte(), 0xCC.toByte(), 0xCE.toByte(), 0xB9.toByte(), 0xC9.toByte(), 0xCB.toByte(), 0xBB.toByte()))
        val canonicalName = simpleBlobProperty.javaClass.canonicalName
        serialize(simpleBlobProperty, canonicalName)
        val out = deserialize(canonicalName, simpleBlobProperty.javaClass)
        System.out.printf("pre %s\n", Arrays.toString(simpleBlobProperty.get()))
        System.out.printf("post %s\n", Arrays.toString(out!!.get()))
    }
}
